src/components/header/Header.tsx

```diff
@@ -69,6 +80,11 @@ export const Header: React.FC = () => {
             iconName="magnifying-glass"
             isActive={searchIsActive}
           />
+          {totalQuantity > 0 ? (
+            <Badge className="badge-icon" label={totalQuantity} />
+          ) : (
+            ''
+          )}
```

Comment:
This code would be more reusable if rather than calculating the total quantity here in the header instead we made a smart component called something like TotalCartQuantityBadge. That badge would get the cart from the context, calculate the total and display it without taking any props. Additionally it could handle the conditional rendering by just returning an empty fragment then the total is 0.